src/qt/votingdialog.cpp

```diff
@@ -885,53 +885,61 @@ void VotingVoteDialog::resetData(const VotingItem *item)
 
 void VotingVoteDialog::vote(void)
 {
-    voteNote_->setStyleSheet("QLabel { color : red; }");
+    // This overall try-catch is needed to properly catch the VoteBuilder builder move constructor and assignment,
+    // otherwise an expired poll bubbles up all the way to the app level and ends execution with the exception handler
+    // in bitcoin.cpp, which is not what is intended here.
+    try {
+        voteNote_->setStyleSheet("QLabel { color : red; }");
 
-    LOCK(cs_main);
+        LOCK(cs_main);
 
-    const PollReference* ref = GetPollRegistry().TryByTxid(pollTxid_);
+        const PollReference* ref = GetPollRegistry().TryByTxid(pollTxid_);
 
-    if (!ref) {
-        voteNote_->setText(tr("Poll not found."));
-        return;
-    }
+        if (!ref) {
+            voteNote_->setText(tr("Poll not found."));
+            return;
+        }
 
-    const PollOption poll = ref->TryReadFromDisk();
+        const PollOption poll = ref->TryReadFromDisk();
 
-    if (!poll) {
-        voteNote_->setText(tr("Failed to load poll from disk"));
-        return;
-    }
+        if (!poll) {
+            voteNote_->setText(tr("Failed to load poll from disk"));
+            return;
+        }
 
-    VoteBuilder builder = VoteBuilder::ForPoll(*poll, ref->Txid());
+        VoteBuilder builder = VoteBuilder::ForPoll(*poll, ref->Txid());
 
-    try {
-        for (int row = 0; row < answerList_->count(); ++row) {
-            if (answerList_->item(row)->checkState() == Qt::Checked) {
-                builder = builder.AddResponse(row);
+        try {
+            for (int row = 0; row < answerList_->count(); ++row) {
+                if (answerList_->item(row)->checkState() == Qt::Checked) {
+                    builder = builder.AddResponse(row);
+                }
             }
+        } catch (const VotingError& e) {
+            voteNote_->setText(e.what());
+            return;
```

Comment:
Want to drop this `try`...`catch`? It looks redundant now.

Reply:
Yep.